src/Functions/FunctionsConversion.h

```diff
@@ -1683,7 +1684,25 @@ struct ConvertImplGenericFromString
 
                 const auto & val = col_from_string->getDataAt(i);
                 ReadBufferFromMemory read_buffer(val.data, val.size);
-                serialization_from.deserializeWholeText(column_to, read_buffer, format_settings);
+                try
+                {
+                    serialization_from.deserializeWholeText(column_to, read_buffer, format_settings);
+                }
+                catch (const Exception & e)
```

Comment:
`ConvertImplGenericFromString` (l. 1644) is a generic handler, putting special handling only for bool types (l. 1691ff.) into it is not beautiful. I wonder how parsing errors of other data types are handled, e.g. ` select accurateCastOrNull('invalid', 'Date');`? (didn't check deeper where the logic is). A better alternative would be to handle errors in bool parsing equivalently to these types.

Reply:
In my previous `Date` example, which code updates the null bitmap on the result column then?

Reply:
Okay then.

Let's address the other comments (about tests) and then we can merge.